### ISSUE TITLE
Entering Game Mode prints an assert to Console

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#9655]
**Describe the bug**
An assert is printed to the Console when entering the Game Mode.

Assert information:
File:
```
/home/gpc-610/builds/o3de/Code/Framework/AzCore/./AzCore/Asset/AssetCommon.h
```
Condition:
```
(140214255760704) 'AZ::Data::Asset<AZ::Data::AssetData>::Asset(const AZ::Data::AssetId &, AZ::Data::AssetData *, AZ::Data::AssetLoadBehavior) [T = AZ::Data::AssetData]'
Asset data already has an ID set.
```
Line: 807

It should be noted that this can be inconsistent, as entering the Game Mode on the same machine this issue was originally noticed did not produce the assert earlier, as can be seen on the video in the https://github.com/o3de/o3de/issues/9653 issue. The Gems included in the project, level used, operating system, or the engine configuration did not seem to affect this issue.

**Steps to reproduce**
1. Enter Game Mode.

**Expected behavior**
Game Mode is entered with no issues.

**Actual behavior**
An assert is printed to the Console.

**Video and Editor.log**

https://user-images.githubusercontent.com/86952082/169295281-8725031d-6f17-4916-8590-7471ac591652.mp4

[Editor.log](https://github.com/o3de/o3de/files/8729101/Editor.log)

**Found in Branch**
Development (e79ad5e)

**Desktop**
 - Device: PC
 - OS: Linux
 - Version: Ubuntu 20.04.3 LTS
 - CPU AMD Ryzen 5 3600
 - GPU NVIDIA GeForce RTX 2060 SUPER
 - Memory 16GB

**Comment by burelc-amzn** [comment]
I can repro this every time:
1. Launch a new instance of the editor
2. Make a new level
3. Delete the Atom Default Environment instance
4. Press Ctrl+G to enter game mode

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Development (0fe387c) build.
